src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs

```diff
@@ -278,26 +293,37 @@ public async Task StartAsync(Uri url, TransferFormat transferFormat, Cancellatio
 
         Log.StartedTransport(_logger);
 
-        // Create the pipe pair (Application's writer is connected to Transport's reader, and vice versa)
-        var pair = DuplexPipe.CreateConnectionPair(_httpConnectionOptions.TransportPipeOptions, _httpConnectionOptions.AppPipeOptions);
+        _stopCts = new CancellationTokenSource();
+
+        var ignoreFirstCanceled = false;
 
-        _transport = pair.Transport;
-        _application = pair.Application;
+        if (_transport is null)
+        {
+            // Create the pipe pair (Application's writer is connected to Transport's reader, and vice versa)
+            var pair = CreateConnectionPair(_httpConnectionOptions.TransportPipeOptions, _httpConnectionOptions.AppPipeOptions);
+
+            _transport = pair.Transport;
+            _application = pair.Application;
+        }
+        else
+        {
+            ignoreFirstCanceled = true;
+        }
 
         // TODO: Handle TCP connection errors
         // https://github.com/SignalR/SignalR/blob/1fba14fa3437e24c204dfaf8a18db3fce8acad3c/src/Microsoft.AspNet.SignalR.Core/Owin/WebSockets/WebSocketHandler.cs#L248-L251
-        Running = ProcessSocketAsync(_webSocket);
+        Running = ProcessSocketAsync(_webSocket, url, ignoreFirstCanceled);
```

Comment:
So this just overwrites the `Running` task when this is called a second time for a seamless reconnect? What happens if `StopAsync` is already awaiting the first one which wich will now stop once this method ends at the end of the first call to `ProcessSocketAsync`?  `StopAsync` would finish while the new connection has already taken over, right? At least the `_stopCts` should still stop everything eventually. I could see this causing problems for the fallback to a normal reconnect though.